qttypes/src/lib.rs

```diff
@@ -875,6 +875,118 @@ mod tests {
     }
 }
 
+cpp_class!(
+    /// Wrapper around [`QVariantMap`][type] typedef.
+    ///
+    /// [type]: https://doc.qt.io/qt-5/qvariant.html#QVariantMap-typedef
+    pub unsafe struct QVariantMap as "QVariantMap"
+);
+
+impl QVariantMap {
+    /// Wrapper around [`insert(int, const QString &, const QVariant &)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qlist.html#insert
+    pub fn insert(&mut self, key: QString, element: QVariant) {
+        cpp!(unsafe [self as "QVariantMap*", key as "QString", element as "QVariant"] {
+            self->insert(key, std::move(element));
+        })
+    }
+
+    /// Wrapper around [`remove(const QString &)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#remove
+    pub fn remove(&mut self, key: QString) -> usize {
+        cpp!(unsafe [self as "QVariantMap*", key as "QString"] -> usize as "size_t" {
+            return self->remove(key);
+        })
+    }
+
+    /// Wrapper around [`take(const QString &)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#take
+    pub fn take(&mut self, key: QString) -> QVariant {
+        cpp!(unsafe [self as "QVariantMap*", key as "QString"] -> QVariant as "QVariant" {
+            return self->take(key);
+        })
+    }
+
+    /// Wrapper around [`size()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#size
+    pub fn len(&self) -> usize {
+        cpp!(unsafe [self as "QVariantMap*"] -> usize as "size_t" {
+            return self->size();
+        })
+    }
+
+    /// Wrapper around [`isEmpty()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#isEmpty
+    pub fn is_empty(&self) -> bool {
+        cpp!(unsafe [self as "QVariantMap*"] -> bool as "bool" {
+            return self->isEmpty();
+        })
+    }
+
+    /// Wrapper around [`contains(const QString &)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qmap.html#contains
+    pub fn contains(&self, key: QString) -> bool {
+        cpp!(unsafe [self as "QVariantMap*", key as "QString"] -> bool as "bool" {
+            return self->contains(key);
+        })
+    }
+}
+
+impl Index<QString> for QVariantMap {
+    type Output = QVariant;
+
+    /// Wrapper around [`at(int)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qlist.html#at
+    fn index(&self, key: QString) -> &QVariant {
+        assert!(self.contains(key.clone()));
+        unsafe {
+            &*cpp!([self as "QVariantMap*", key as "QString"] -> *const QVariant as "const QVariant*" {
```

Comment:
I'd return `-> Option<&QVariant> as "const QVariant*"`  
and then do a `.expect("key not in the QVariant")`  instead of the assert.  that way you only do the lookup once.

You might also need a `#[track_caller]` so the panic location appears for the user who uses that function

Reply:
Using `.find().value()` seems to return a non-null garbage pointer.

Reply:
```suggestion
            &*cpp!([self as "QVariantMap*", key as "QString"] -> *const QVariant as "const QVariant*" {
```

Actually, you must use `self as "const QVariantMap*"`   or `constFind`

Reply:
(and then compare the result with constEnd